client/app/accounts/services/account-guard.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot
+} from '@angular/router';
+import { CurrentAccountService } from './current-account.service';
+
+@Injectable()
+export class AccountGuardService implements CanActivate {
+
+  constructor(
+    private currentAccountService: CurrentAccountService,
+    private router: Router) { }
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+    const url: string = state.url;
+    return this.checkLogin(url);
+  }
+
+  checkLogin(url: string): Observable<boolean> {
+    return this.currentAccountService.get().map((account: string) => {
+      if (! account) {
+        // Store the attempted URL for redirecting after login
+        this.currentAccountService.redirectUrl = url;
```

Comment:
I don't like this implementation, the service should not store this information. I think we rather should add it to the login query string (somehting like http://localhost:4200/accounts/login?redirectUrl=%2Fprofile)